### PR TITLE
Feature/mental health liner questions

### PR DESCRIPTION
**Pull request by petermason**
# Description

Allow the user to answer one question at a time.
animate (fade) next question in
activate submit once all questions answered

## On what platform have you tested the change?

- [x] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

https://user-images.githubusercontent.com/795789/107804238-925dbb80-6d5b-11eb-9f53-7921dca19054.mov

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

**Comment by selvach** [comment]
WOW. This is beautiful 👍

File: src/features/dashboard/DashboardScreen.tsx [hunk 1]
```diff
@@ -74,7 +74,10 @@ export const DashboardScreen: React.FC<Props> = ({ navigation, route }) => {
   };
 
   const showMentalHealthModal = () => {
-    appCoordinator.goToMentalHealthModal();
+    console.log('consent: ', MentalHealthState.consent);
```
Comment (selvach) [inline src/features/dashboard/DashboardScreen.tsx:77]: Stray console.log

File: src/features/mental-health/screens/MentalHealthModal.tsx [hunk 2]
```diff
@@ -46,17 +55,21 @@ function MentalHealthModal() {
           </Text>
         </View>
         <View>
-          <TouchableOpacity style={[styles.button, { backgroundColor: '#0165B5' }]} onPress={handleSetConsent}>
+          <TouchableOpacity
+            style={[styles.button, { backgroundColor: '#0165B5' }]}
```
Comment (selvach) [inline src/features/mental-health/screens/MentalHealthModal.tsx:59]: Would be nice to get these colours into a theme file / typed object

File: src/features/dashboard/DashboardScreen.tsx [hunk 3]
```diff
@@ -74,7 +74,9 @@ export const DashboardScreen: React.FC<Props> = ({ navigation, route }) => {
   };
 
   const showMentalHealthModal = () => {
-    appCoordinator.goToMentalHealthModal();
+    if (MentalHealthState.consent !== 'NO') {
```
Comment (bionara) [inline src/features/dashboard/DashboardScreen.tsx:77]: maybe should be a const or enum

File: src/features/mental-health/partials/questions/changes-question.tsx [hunk 4]
```diff
@@ -21,9 +22,9 @@ type TAnswer = {
   keyValue: TKeyValue;
 };
 
-function ChangesQuestion({ onPress, question, state }: IProps) {
+function ChangesQuestion({ disabled = false, onPress, question, state }: IProps) {
   const { grid } = useTheme();
-
+  const fadeAnim = useFade(0.2, disabled ? 0.2 : 1, 500);
```
Comment (bionara) [inline src/features/mental-health/partials/questions/changes-question.tsx:27]: would be good to have this fade conf - alpha and duration - as a const somewhere

File: src/features/mental-health/screens/MentalHealthModal.tsx [hunk 5]
```diff
@@ -1,17 +1,26 @@
...
 import appCoordinator from '../../AppCoordinator';
 
 function MentalHealthModal() {
+  const dispatch = useDispatch();
   const { goBack } = useNavigation();
 
   // TODO implement redux state
-  const handleSetConsent = () => {
+  const handleSetConsent = (consent: TMentalHealthConsent) => {
+    dispatch(setConsent(consent));
```
Comment (bionara) [inline src/features/mental-health/screens/MentalHealthModal.tsx:19]: cnst/enum

File: src/features/mental-health/screens/MentalHealthFrequency.tsx [hunk 6]
```diff
@@ -59,8 +62,10 @@ function MentalHealthFrequency() {
         </Text>
         {questions.map((item, index) => {
           const key = `changes-${index}`;
+          const disabled = index > curQuestion;
```
Comment (bionara) [inline src/features/mental-health/screens/MentalHealthFrequency.tsx:65]: just from the code - will this gave any undesired effects if i "go back", ie a previous question is set to current?

File: src/features/mental-health/screens/MentalHealthModal.tsx (hunk 2 above)
Comment (bionara) [reply src/features/mental-health/screens/MentalHealthModal.tsx:59]: same for the 'Later', 'No', and fade conf.

**Review by bionara — APPROVED** [review]

File: src/features/mental-health/screens/MentalHealthFrequency.tsx (hunk 6 above)
Comment (petermason) [reply src/features/mental-health/screens/MentalHealthFrequency.tsx:65]: no, state is managed via redux not the component

File: src/features/mental-health/screens/MentalHealthModal.tsx (hunk 5 above)
Comment (petermason) [reply src/features/mental-health/screens/MentalHealthModal.tsx:19]: cnst?

File: src/features/mental-health/partials/questions/changes-question.tsx (hunk 4 above)
Comment (petermason) [reply src/features/mental-health/partials/questions/changes-question.tsx:27]: plan to add this to the theme in future pr

**Pull request merged**